File: bench/native-speed/zig/intsum.zig
const std = @import("std");

pub fn main() void {
    const n: i64 = 1_000_000_000;
    const md: i64 = 1_000_000_007;
    var s: i64 = 0;
    var i: i64 = 1;
    while (i <= n) : (i += 1) {
        s = @rem(s + i * i, md);
    }
    std.debug.print("{d}\n", .{s});
}

File: bench/native-speed/rust/intsum.rs
fn main() {
    let n: i64 = 1_000_000_000;
    let md: i64 = 1_000_000_007;
    let mut s: i64 = 0;
    let mut i: i64 = 1;
    while i <= n {
        s = (s + i * i) % md;
        i += 1;
    }
    println!("{}", s);
}
